include/deal.II/fe/fe_system.h

```diff
@@ -1056,6 +1049,39 @@ class FESystem : public FiniteElement<dim, spacedim>
   virtual std::size_t
   memory_consumption() const override;
 
+  /**
+   * Computing values on a cell requires copying values between the
+   * FEValuesImplementation::FiniteElementRelatedData objects owned by the
+   * present object and its subelements. The indexing for this is fairly
+   * complicated since each shape function in the nonprimitive base element may
+   * have a different number of nonzero components (e.g., a Taylor-Hood pair
+   * base element): we also need offsets into both the base element and present
+   * element's arrays. To this end, we will create a table with all the
+   * necessary information for each shape function in the nonprimitive base
+   * element.
+   */
+  struct BaseOffsets
+  {
+    /**
+     * Number of nonzero components for the current base element shape function.
+     */
+    unsigned int n_nonzero_components;
+
+    /**
+     * Index into the base element's
+     * FEValuesImplementation::FiniteElementRelatedData arrays for the current
+     * base element shape function.
+     */
+    unsigned int in_index;
+
+    /**
+     * Index into the present element's
+     * FEValuesImplementation::FiniteElementRelatedData arrays for the current
```

Comment:
this comment is identical to above, copy&paste mistake?

Reply:
It isn't identical: this is for the present element, the first is for the base element (one word is different).